services/clientlog/pkg/service/service.go

```diff
@@ -228,13 +201,78 @@ func processFileEvent(ctx context.Context, ref *provider.Reference, gwc gateway.
 	return users, data, err
 }
 
-// adds userid to users slice or gets members of groupid and adds them to users slice
-func addSharees(ctx context.Context, users []string, gwc gateway.GatewayAPIClient, uid *user.UserId, gid *group.GroupId) ([]string, error) {
+// process share related events
+func processShareEvent(ctx context.Context, ref *provider.Reference, gwc gateway.GatewayAPIClient, initiatorid string, shareeID *user.UserId, shareeGroupID *group.GroupId) ([]string, FileEvent, error) {
+	users, data, err := processFileEvent(ctx, ref, gwc, initiatorid)
+	if err != nil {
+		return users, data, err
+	}
+
+	return addShareeData(ctx, gwc, data, users, shareeID, shareeGroupID)
+}
+
+// custom logic for item trashed event
+func processItemTrashedEvent(ctx context.Context, ref *provider.Reference, gwc gateway.GatewayAPIClient, initiatorid string, itemID *provider.ResourceId) ([]string, FileEvent, error) {
+	resp, err := gwc.ListRecycle(ctx, &provider.ListRecycleRequest{
+		Ref: ref,
+	})
+	if err != nil {
+		return nil, FileEvent{}, err
+	}
+	if resp.GetStatus().GetCode() != rpc.Code_CODE_OK {
+		return nil, FileEvent{}, fmt.Errorf("error listing recycle: %s", resp.GetStatus().GetMessage())
+	}
+
+	for _, item := range resp.GetRecycleItems() {
+		if item.GetKey() == itemID.GetOpaqueId() {
+
+			data := FileEvent{
+				ItemID: storagespace.FormatResourceID(*itemID),
+				// TODO: check with web if parentID is needed
+				// ParentItemID: storagespace.FormatResourceID(*item.GetRef().GetResourceId()),
+				SpaceID:     storagespace.FormatStorageID(itemID.GetStorageId(), itemID.GetSpaceId()),
+				InitiatorID: initiatorid,
+			}
+
+			users, err := utils.GetSpaceMembers(ctx, itemID.GetSpaceId(), gwc, utils.ViewerRole)
```

Comment:
Shouldn't we return an empty `FileEvent{}`  if the `GetSpaceMembers` returns an error?

Reply:
Yeah, good point. Would be something for the coding guidelines we talked about :)

I wanted to avoid the unnecessary error check for better readability. Compare:
```suggestion
			users, err := utils.GetSpaceMembers(ctx, itemID.GetSpaceId(), gwc, utils.ViewerRole)
			if err != nil {
			    return nil, FileEvent{}, err
			}
			return users, data, nil
```
versus:
```suggestion
			users, err := utils.GetSpaceMembers(ctx, itemID.GetSpaceId(), gwc, utils.ViewerRole)
			return users, data, err
```
Imho if you do not intend to do something special, you do not need to explicitly check the error. But that is up for debate.